src/hooks/useBankTransactions/useAugmentedBankTransactions.tsx

```diff
@@ -192,19 +192,24 @@ export const useAugmentedBankTransactions = (
     return filtered
   }, [filters, data])
 
-  const updateOneLocal = (newBankTransaction: BankTransaction) => {
-    if (newBankTransaction.update_categorization_rules_suggestion) {
-      const decodedRuleSuggestion = decodeRulesSuggestion(newBankTransaction.update_categorization_rules_suggestion)
-      setRuleSuggestion(decodedRuleSuggestion)
-    }
-    const updatedData = rawResponseData?.map((page) => {
-      return {
-        ...page,
-        data: page.data?.map(bt =>
-          bt.id === newBankTransaction.id ? newBankTransaction : bt,
-        ),
+  const updateLocalBankTransactions = (newBankTransactions: BankTransaction[]) => {
+    const transactionsById = new Map(
+      newBankTransactions.map(bt => [bt.id, bt]),
+    )
+
+    for (const bt of newBankTransactions) {
+      if (bt.update_categorization_rules_suggestion) {
+        const decodedRuleSuggestion = decodeRulesSuggestion(bt.update_categorization_rules_suggestion)
+        setRuleSuggestion(decodedRuleSuggestion)
       }
-    })
+    }
+
+    const updatedData = rawResponseData?.map(page => ({
+      ...page,
+      data: page.data?.map(bt =>
+        transactionsById.has(bt.id) ? transactionsById.get(bt.id)! : bt,
```

Comment:
No need for non-null assertion here.
```suggestion
        transactionsById.has(bt.id) ?? bt,
```